### PR TITLE
Basic install of CP4WA

### PR DESCRIPTION
**Pull request by doormat18**
Signed-off-by: doormat18 <mat.davis@gmail.com>

### CODE REVIEW

File: automation-generators/generic/cp4wa/preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
+from generatorPreProcessor import GeneratorPreProcessor
```
Comment (fketelaars) [inline automation-generators/generic/cp4wa/preprocessor.py:1]: This preprocessor  pertains to cp4d. If you don't have a preprocessor yet, pease remove the .py file and no linting will take place.

File: automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/templates/cp4wa-catalog-sources.j2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (fketelaars) [inline automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/templates/cp4wa-catalog-sources.j2:1]: For CP4D and CP4I we have stepped away from using the ibm-operator-catalog and plan to only support installation through case files to have a specific release. I would recommend to do the same for CP4WA.

File: automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/vars/main.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+implemented_cloud_platform_types: ["existing-ocp"]
```
Comment (fketelaars) [inline automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/vars/main.yaml:2]: Does the CP4WA automation only support existing-ocp?

**Review by fketelaars — CHANGES_REQUESTED** [review]
Looks good overall, just a few comments. Before merging with `main` I would like to merge into the `cp4wa` branch and run through a CP4D installation to test regression.

File: automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/templates/cp4wa-catalog-sources.j2 (hunk 2 above)
Comment (doormat18) [reply automation-roles/50-install-cloud-pak/cp4wa/cp4wa-cluster/templates/cp4wa-catalog-sources.j2:1]: Can I create an issue for this and add to the next iteration? Are you saying we should remove the option for ibm-operator-catalog as well?

File: automation-generators/generic/cp4wa/preprocessor.py (hunk 1 above)
Comment (doormat18) [reply automation-generators/generic/cp4wa/preprocessor.py:1]: That preprocessor expands the OCS storage to the actual StorageClasses. The script fails if I remove it.

Comment (doormat18) [reply automation-generators/generic/cp4wa/preprocessor.py:1]: I can look at the file and try and strip out some of the unnecessary stuff if you'd like.

**Review by fketelaars — APPROVED** [review]

**Pull request merged**